controllers/agentcluster_controller_test.go

```diff
@@ -147,7 +147,8 @@ var _ = Describe("agentcluster reconcile", func() {
 	)
 
 	BeforeEach(func() {
-		c = fakeclient.NewClientBuilder().WithScheme(scheme.Scheme).Build()
+		agentCluster := &capiproviderv1.AgentCluster{}
+		c = fakeclient.NewClientBuilder().WithScheme(scheme.Scheme).WithStatusSubresource(agentCluster).Build()
```

Comment:
For history, had to add `WithStatusSubresource` to this because of https://github.com/kubernetes-sigs/controller-runtime/issues/2362 which prevents the status of a resource ffrom being updated during unit tests, causing them to fail.
Change ref: https://github.com/evryfs/github-actions-runner-operator/commit/165da33dd0cdd7ab85dff76f9bc6b851b8bb4a1f